lib/alerts/medicine_article_monitor.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+# This class to create alerts when a WikiProject Medicine article
+# is assigned for a course that doesn't have the medical content
+# training module.
+# wiki_education mode only
+# Goal: to catch cases where a class that wasn't expected 
+# to work on medical topics tries to do so.
+# Issue #4693
+class MedicineArticleMonitor
+  WIKI_PROJECT_MEDICINE_CAT = 'Category:All WikiProject Medicine articles'
+  NUMBER_OF_ARTICLES_TO_CHECK = 5
+
+  def self.create_alerts_for_no_med_training_for_course
+    new.create_alerts_for_no_med_training_for_course
+  end
+
+  def initialize
+    @wiki = Wiki.find_by(language: 'en', project: 'wikipedia')
+  end
+
+  def create_alerts_for_no_med_training_for_course
+    last_assigned_med_articles.each do |assignment|
+      unless med_training_for_course?(assignment.course_id)
+        create_alert_for_no_med_training_for_course(assignment)
+      end
+    end
+  end
+
+  def create_alert_for_no_med_training_for_course(assignment)
+    return if alert_already_exists?(assignment)
+
+    alert = Alert.create!(type: 'NoMedTrainingForCourseAlert',
+                  article_id: assignment.article_id,
+                  course_id: assignment.course_id)
+    alert.email_content_expert
+  end
+
+  def alert_already_exists?(assignment)
+    Alert.exists?(type: 'NoMedTrainingForCourseAlert',
+                  article_id: assignment.article_id,
+                  course_id: assignment.course_id)
+  end
+
+  def last_assigned_med_articles
+    last_assigned_articles
+      .select { |assignment| med_article?(assignment.article_title) }
+  end
+
+  def last_assigned_articles
+    Assignment.last NUMBER_OF_ARTICLES_TO_CHECK
```

Comment:
I don't think looking at a fixed number of recent assignments is the right strategy here. There will be times when a large number of assignments are added between two consecutive `constant update` cycles.

We want to start generating alerts for new assignments, without adding a slew of useless alerts for courses from the past. Finding recent assignments based on creation date (eg, assignments created within the last day) is probably a good approach.

Reply:
From reading from `config/schedule.yml`, I read that ConstantUpdateWorker is executed each 4 minutes.
This is why I have chosen a fixed number. 

If I take daily created assignements, there will be a lot of trying to create alerts.
Though I check that an alert does not exists before creating it.
I will do that: the last 24 hours.
But, to be sure, the cron run each 4 minutes ?

Reply:
Yes, the cron runs every 4 minutes at most. It won't schedule a new one if the previous one is still running, though (because of the `lock: :until_executed` option for that worker, which is part of the `sidekiq_unique_jobs` plugin).

I think a shorter time than 24 hours would be okay, but I'm guessing that making it that long won't cause any problems and will have only a minor impact on execution time.